wurst/systems/spells/beastMaster/Shapeshift.wurst

```diff
@@ -0,0 +1,141 @@
+package Shapeshift
+import AbilityObjEditing
+import ObjectIdGenerator
+import ID
+import ClosureEvents
+import HashMap
+import ObjectIds
+import MeatSystem
+import SimError
+import LastOrder
+import ClosureTimers
+
+let FORM_IDS = [UNIT_SHAPESHIFTER_WOLF, UNIT_SHAPESHIFTER_BEAR, UNIT_SHAPESHIFTER_PANTHER, UNIT_SHAPESHIFTER_TIGER]
+let FORM_MAP = new HashMap<string, integer>()
+let TRANSFORM_ABILITY_IDS = [compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()),
+compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()),
+compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next()), compiletime(ABIL_ID_GEN.next())]
+public let ABILITY_ID_VICIOUS_STRIKES_ORB = 'A0GG'
+public let ABILITY_ID_VICIOUS_STRIKES_ACTIVE = 'A0GH'
+public let ABILITY_ID_BEARS_BULWARK = 'A0GE'
+public let ABILITY_ID_WOLFS_HUNGER = 'A0GD'
+public let ABILITY_ID_PROWL = 'A0GF'
+let FORM_SPELLBOOK = 'A0EP'
+let ABILITY_ID_WOLF_FORM_DUMMY = 'A00H'
+let ABILITY_ID_BEAR_FORM_DUMMY = 'A0GN'
+let ABILITY_ID_PANTHER_FORM_DUMMY = 'A0GO'
+let ABILITY_ID_TIGER_FORM_DUMMY = 'A0GP'
+let TRANSFORM_FX_PATH = "Abilities\\Spells\\Human\\Polymorph\\PolyMorphDoneGround.mdl"
+
+@compiletime function formChanger()
+    var i = 0
+    for oldForm = 0 to 3
+        for newForm = 0 to 3
+            if newForm != oldForm
+                new AbilityDefinitionBearform(TRANSFORM_ABILITY_IDS[i])
+                ..setName(FORM_IDS[oldForm].toRawCode() + " to " + FORM_IDS[newForm].toRawCode())
+                ..presetAlternateFormUnit(lvl -> FORM_IDS[oldForm].toRawCode())
+                ..presetNormalFormUnit(lvl -> FORM_IDS[newForm].toRawCode())
+                i++
+
+function onCast(int newFormId)
+    var caster = GetSpellAbilityUnit()
+    var owner = caster.getOwner()
+    var oldF = caster.getTypeId()
+    var changeKey = oldF.toRawCode() + newFormId.toRawCode()
+
+    //Re-issue whatever order we had previously
+    nullTimer() ->
+        let previousOrder = caster.getLastOrder().prev
```

Comment:
Can you not base the shapeshift abilities on a dummy windwalk so you can use them while walking?